docs/guide.rst

```diff
@@ -31,109 +31,179 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
+automatically selects an appropriate backend to talk to the GPU and the OS::
 
-    # The actual rendering
-    renderer.render(scene, camera)
+    # @almarklein: Can we provide a more meaningful example
+    import pygfx as gfx
+    from wgpu.gui.auto import WgpuCanvas
 
+    canvas = WgpuCanvas()
+    cube = gfx.Mesh(
+        gfx.box_geometry(200, 200, 200),
+        gfx.MeshPhongMaterial(color="#336699"),
+    )
 
-Putting it together
--------------------
+    if __name__ == "__main__":
+        gfx.show(cube, canvas=canvas)
 
-If you run this, you should see a rotating yellow cube.
+.. image:: _static/guide_hello_world.png
 
-.. code-block:: python
+If you run the above, you will notice that it looks exactly the same as our
+introductory example. Just as with a `Scene` above, `gfx.show` will create a
+`Canvas` for you unless you provide it explicitly, and by default this uses the
+``auto`` backend and hence we won't see any changes.
 
-    import pygfx as gfx
+Why then would we explicitly provide a `Canvas`? Well, beyond the `Canvas`
+itself a backend also provides an event loop. This event loop allows us to
+animate the scene or build a GUI by registering callbacks. To ensure that this
+works correctly we need to use the same backend for both the event loop and the
+`Canvas` and, for more complex scenarios, we may thus wish to create the
+`Canvas` explicitly.
 
-    from PySide6 import QtWidgets
-    from wgpu.gui.qt import WgpuCanvas
+**Renderers**
 
+The third and final main building block is a `Renderer`. A `Renderer` is like an
+artist that brings all of the above together. It looks at the `Scene` through a
+`Camera` and draws what it sees onto the surface provided by the `Canvas`. Like
+any good artist, a `Renderer` is never seen without its `Canvas`, so to create a
+`Renderer` we also need to create a `Canvas`::
 
-    app = QtWidgets.QApplication([])
+    # @almarklein: Can we provide a more meaningful example
+    import pygfx as gfx
+    from wgpu.gui.auto import WgpuCanvas
 
-    # Create a canvas and a renderer
     canvas = WgpuCanvas()
     renderer = gfx.renderers.WgpuRenderer(canvas)
+    
+    cube = gfx.Mesh(
+        gfx.box_geometry(200, 200, 200),
+        gfx.MeshPhongMaterial(color="#336699"),
+    )
+
+    if __name__ == "__main__":
+        gfx.show(cube, renderer=renderer)
+
+.. image:: _static/guide_hello_world.png
+
+As you may expect by now, the output is the same as without the explicit
+reference because `gfx.show` will create a renderer if you don't provide it. For
+many applications this is perfectly fine; however, if we want to tackle more
+advanced problems (e.g., control the exact process on how objects appear to
+overlay each other) we may need to create it explicitly. For starters, it is
+enough to know that it exists and what it does, so that we can come back to it
+later when it becomes relevant.
+
+(@almarklein: Do we have examples that customize the renderer that we could
+reference here?)
```

Comment:
@almarklein Hm, I was thinking about this as well but was wondering if this isn't too advanced a subject for a newcomer. The same for the blend mode. It's an excellent example, but I feel it would be better placed into a dedicated tutorial on blend modes and why they are important.

Re-reading the guide now, I think it could be fine to have the example as is. How do you think?

Reply:
Yeah, mentioning that the renderer allows advanced tuning of the rendering process, like you do now, is fine! 👍